### PR TITLE
refactor: extract colorful domain types

### PR DESCRIPTION
**Pull request by ryota-murakami**
## Summary
- Extract shared domain aliases for counts, byte sizes, progress, file metadata, dashboard grid cells, and settings patches.
- Thread `SettingsPatch` through IPC/preload/main/renderer without changing the runtime Zod schema or persisted shape.
- Brand `SymlinkCleanupItemId` at factory boundaries and preserve branded IDs through cleanup retry/error filtering.
- Keep single-use helper aliases local to avoid generic alias soup.

## Verification
- pnpm typecheck
- pnpm lint
- pnpm exec vitest run src/renderer/src/components/dashboard/utils/buildSymlinkCleanupPlan.test.ts src/main/ipc/ipc-schemas.test.ts src/main/services/settings.test.ts
- pnpm validate

## Review Notes
- Subagent cross-review found one row-error edge case; fixed by preserving visible row errors by key presence rather than message truthiness.
- Runtime behavior, IPC wire shape, Redux persisted shape, and filesystem behavior are intended to remain unchanged.

## Summary by CodeRabbit

* **Refactor**
  * 共有型の大幅整理：数値/ファイル単位の型別名追加、ダッシュボード座標型導入、各種ペイロードの型強化。設定パッチ用の公開型（SettingsPatch）を追加し書き込み契約を統一。
* **Bug Fixes**
  * ウィジェット一覧のプレビュー/フォーカス挙動を簡素化。
  * シンボリックリンククリーンアップ画面で壊れた行表示とエラー絞り込みを改善し視認性向上。
* **Tests**
  * クリーンアップ行IDの安定性を検証するテストを追加。

### CODE REVIEW

File: src/renderer/src/components/dashboard/WidgetPicker.tsx [hunk 1]
```diff
@@ -173,19 +173,12 @@ export const WidgetPicker = React.memo(function WidgetPicker({
                     type="button"
                     ref={isSeedRow ? seedRowRef : undefined}
                     onClick={() => handleAddWidget(widgetDefinition.type)}
-                    onMouseEnter={() =>
+                    onPointerMove={() => {
+                      if (activePreviewType === widgetDefinition.type) return
                       dispatch(setActivePreviewType(widgetDefinition.type))
-                    }
+                    }}
```
[the remark this answers is not in the document]
Comment (ryota-murakami) [reply src/renderer/src/components/dashboard/WidgetPicker.tsx:179]: Keeping `onPointerMove` intentionally here. This PR hit a CI browser flake where a dialog opening under a stationary pointer let the mount-time hover row steal the seeded preview; switching back to `onPointerEnter` can reintroduce that path because enter-style hover can fire as the row appears under the pointer. The current handler keeps hover movement responsive, adds the `activePreviewType` equality guard to avoid redundant Redux dispatches, and preserves the keyboard-focus preview path. Verified with 5x `WidgetPicker.browser.test.tsx` loop and `pnpm validate`.
